core/src/main/resources/hudson/diagnosis/ReverseProxySetupMonitor/resources.js

```diff
@@ -9,13 +9,16 @@
 
   var urlToTest = redirectForm.getAttribute("data-url");
   var callUrlToTest = function (testWithContext, callback) {
-    var options = {
-      onComplete: callback,
-    };
+    var body = null;
     if (testWithContext === true) {
-      options.parameters = { testWithContext: true };
+      body = new URLSearchParams({ testWithContext: "true" });
     }
-    new Ajax.Request(urlToTest, options);
+    fetch(urlToTest, {
+      cache: 'no-cache',
+      headers: crumb.wrap({}),
+      body
+    }).then(rsp => callback(rsp))
+      .catch(rsp => callback(rsp))
```

Comment:
> so what is the need for the catch line

https://developer.mozilla.org/en-US/docs/Web/API/fetch
> A [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/fetch) promise only rejects when a network error is encountered (which is usually when there's a permissions issue or similar). A [fetch()](https://developer.mozilla.org/en-US/docs/Web/API/fetch) promise does not reject on HTTP errors (404, etc.). Instead, a then() handler must check the [Response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok) and/or [Response.status](https://developer.mozilla.org/en-US/docs/Web/API/Response/status) properties.

This admin monitor is checking for incorrect configuration / networking issues, and as such when your Jenkins / reverse proxy / whatever is not setup correctly this can (and does) cause promise rejections in this code.

----

> by invoking the callback twice if the callback throws an exception (which the old code didn't do)

It could although all the callback does it show a warning message so I think should be okay for now?

Reply:
Thank you, this makes sense. Since in all other places in this diff the common convention is to not catch promise rejections, the reader of this code may be naturally wondering why this does not match the general paradigm. Since the exception to the rule is justified here, I think it would be worth adding a code comment summarizing the justification for this exception which you have so clearly written above.